enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,193 @@
+computed might outweight the added value of virtualisation.
+
+In OpenStack clusters where Nova transparently provisions baremetal machines,
+this enhancement will ensure that OpenShift installations can use them as
+Compute Nodes.
+
+### Goals
+
+* Enable running OpenShift on a mix of virtual and baremetal machines.
+* Baremetal machines can be added and removed from a running OpenShift cluster.
+
+### Non-Goals
+
+**Separate subnets.**
+
+An OpenStack setting where baremetal machines can't be attached to a virtual
+subnet is not supported in this enhancement.
+
+**External connectivity via floating IPs.**
+
+The OpenStack cluster might make the baremetal nodes accessible, but exposing
+them via floating IPs might represent an increase in complexity and is
+currently not a goal.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+* The choice between deploying a virtual or baremetal machine is done by selecting an appropriate Nova flavour
+  * which can be done both in install config and passed to Cluster API provider OpenStack
+  * for the time being, the interactive prompt does not ask the flavor for the Compute Nodes specifically; the choice must be made editing `install-config.yaml`
+* The deployment request makes the same API call to Nova as with a virtual machine install (there are no explicit calls to Ironic or anything in the Installer required)
+  * so creating baremetal nodes should work transparently from the installer as well as Cluster API provider OpenStack
+* both virtual and baremetal machines consume the same image from glance
+
+### Probable issues
+
+* When attaching and detaching, the expected timings must be adjusted to
+  reflect the different boot and shutdown latencies of physical machines
+
+* When attaching, the Certificate Signing Request timeout must be adjusted to
+  reflect the different boot and shutdown latencies of physical machines
+
+### User Stories
+
+#### Increased performance
+
+User wants to run the Compute Nodes on baremetal machines for increased
+performance. The Control Plane sits on regular Nova instances to leverage the
+flexibility of virtual machines.
+
+### Risks and Mitigations
+
+**Uncertainty about OpenStack's extra networking resources**
+
+Are ports (including trunk), security groups, allowed address pairs all these necessary for baremetal?
+
+Will the Cluster-API-Provider be able to set them on baremetal-based nodes? Is it even possible to assign a port or a security group to a baremetal machine?
+
+A process where nodes are manually created and attached, such as what described in the UPI documentation, should provide a mitigation.
```

Comment:
Could this be elaborated? This feels like an entirely different user story.

Reply:
It's a mitigation if we can't reasonably get the installer and cluster-api-provider-openstack to create the bare metal nodes transparently.

You could create and add them to an existing OpenShift cluster manually. The UPI docs describe how to do a fully-manual deployment of OpenShift on OpenStack. I'll elaborate on this section and link the UPI doc.